Source/JavaScriptCore/wasm/WasmOperations.cpp

```diff
@@ -829,6 +830,42 @@ JSC_DEFINE_JIT_OPERATION(operationWasmRefFunc, EncodedJSValue, (Instance* instan
     return JSValue::encode(value);
 }
 
+JSC_DEFINE_JIT_OPERATION(operationWasmStructNew, EncodedJSValue, (Instance* instance, uint32_t typeIndex, uint64_t* arguments))
+{
+    JSWebAssemblyInstance* jsInstance = instance->owner<JSWebAssemblyInstance>();
+    Ref<TypeDefinition> structTypeDefinition = jsInstance->instance().module().moduleInformation().typeSignatures[typeIndex];
+    const StructType& structType = *structTypeDefinition->as<StructType>();
+    JSGlobalObject* globalObject = jsInstance->globalObject();
+    VM& vm = globalObject->vm();
+
+    FixedVector<uint64_t> values(structType.fieldCount());
+    for (unsigned i = 0; i < structType.fieldCount(); ++i)
+        values[i] = arguments[i];
+
+    JSValue structValue = JSWebAssemblyStruct::create(globalObject, vm, globalObject->webAssemblyStructStructure(), Wasm::WasmStruct::create(WTFMove(structTypeDefinition), WTFMove(values)));
```

Comment:
Should this just take in a global object?

Reply:
Thanks, I've removed a few parameters that can be obtained through global obj.